ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -389,6 +389,14 @@ KEXT_STATIC VirtualizationRootHandle InsertVirtualizationRoot_Locked(PrjFSProvid
     {
         assert(rootIndex < s_maxVirtualizationRoots);
         assert(!s_virtualizationRoots[rootIndex].inUse);
+        
+        // Retain a strong reference to the vnode if we have an active provider on it
```

Comment:
Is the main reason for doing this here (rather than in `VirtualizationRoot_RegisterProviderForPath`) that it makes the unit tests cleaner?

Reply:
Having `InsertVirtualizationRoot_Locked` pair nicely with `ActiveProvider_Disconnect()` also is a necessary requirement for the `s_virtualizationRoots` array resize tidyup that was part of #500 together with this fix. I'm going to pull that cleanup out into a separate PR as well once the higher priority stuff has been dealt with.